images/cluster-setup/register.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The pipelines-service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+
+    printf "Usage: DATA_DIR="/workspace" ./register.sh \n\n"
+
+    # Parameters
+    printf "The following parameters need to be passed to the script: \n"
+    printf "DATA_DIR: the location of the cluster files \n"
+}
+
+prechecks () {
+    DATA_DIR=${DATA_DIR:-}
+    if [[ -z "${DATA_DIR}" ]]; then
+        printf "DATA_DIR not set\n\n"
+        usage
+	      exit 1
+    fi
+}
+
+# populate clusters with the cluster names taken from the kubeconfig
+# populate contexts with the context name taken from the kubeconfig
+# populate kubeconfigs with the associated kubeconfig for each cluster name
+# only consider the first context for a specific cluster
+get_clusters() {
+    clusters=()
+    contexts=()
+    kubeconfigs=()
+    printf "Extracting files under the kubeconfig dir and reading the content in each file \n"
+    files=("$(ls "$DATA_DIR/gitops/credentials/kubeconfig/compute")")
+    for kubeconfig in "${files[@]}"; do
+        printf "%s\n" "$kubeconfig"
+        subs=("$(KUBECONFIG=${DATA_DIR}/gitops/credentials/kubeconfig/compute/${kubeconfig} kubectl config view -o jsonpath='{range .contexts[*]}{.name}{","}{.context.cluster}{"\n"}{end}')")
+        for sub in "${subs[@]}"; do
+            context=$(echo -n "${sub}" | cut -d ',' -f 1)
+            cluster=$(echo -n "${sub}" | cut -d ',' -f 2 | cut -d ':' -f 1)
+	    if ! (echo "${clusters[@]}" | grep "${cluster}"); then
+                clusters+=( "${cluster}" )
+                contexts+=( "${context}" )
+                kubeconfigs+=( "${kubeconfig}" )
+                printf "%s --- %s --- %s \n" "$cluster" "$context" "$kubeconfig"
+            fi
+        done
+    done
+
+}
+
+gitops_installed_check() {
+#  printf "Checking if Openshift GitOps is installed on the cluster... \n"
+    for i in "${!clusters[@]}"; do
+      #checking if the gitops server pod is up and running
+      declare podname=""
+      declare -i i=0
+      declare -i timeout=60
+      printf "Checking if the GitOps cluster pod is available and Ready."
```

Comment:
I am not sure that it is the best check.
The pod may be temporarily  unavailable (due to an upgrade, due to a node maintenance) and still GitOps installed and working.
The operator resource may be a better fit

Reply:
Yeah, I had done operator check first, but then had the same thought that the operator resource may be up, but the pod might not be :)
I can do both the checks if that is better.

Besides, I think if you compare just operator vs pods, pods are almost always alive even during upgrades (as k8s takes care of this to always serve traffic).